### PR TITLE
Bug 226103 - findMouseEventTargetAt problem with local coordinate system

### PR DESCRIPTION
**Pull request by ptziegler**
A figure using local coordinates translates the given point for its children at the beginning of findMouseEventTargetInDescendantsAt(int, int). However, during iteration over its children, the translated point is stored in a  static variable PRIVATE_POINT. The value would be changed by a child, if the child also uses local coordinates. This affects the following queries to the rest of children, if no figure is found under that child.

Fixes https://github.com/eclipse/gef-classic/issues/495

### CODE REVIEW

File: org.eclipse.draw2d.tests/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -2,14 +2,17 @@ Manifest-Version: 1.0
...
 Bundle-SymbolicName: org.eclipse.draw2d.tests
-Bundle-Version: 3.13.100.qualifier
+Bundle-Version: 3.14.0.qualifier
+Import-Package: org.slf4j
 Bundle-Vendor: %Plugin.providerName
 Bundle-Localization: plugin
 Require-Bundle: org.eclipse.core.runtime;bundle-version="[3.2.0,4.0.0)",
  org.eclipse.core.resources;bundle-version="[3.2.0,4.0.0)",
  org.eclipse.draw2d;bundle-version="[3.16.0,4.0.0)",
+ org.eclipse.draw2d.examples;bundle-version="3.15.400",
```
Comment (azoitl) [inline org.eclipse.draw2d.tests/META-INF/MANIFEST.MF:12]: shouldn't this be 3.16.0 and should we really be so explicit?

File: org.eclipse.draw2d.tests/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -2,14 +2,17 @@ Manifest-Version: 1.0
...
  org.junit,
- org.eclipse.jface;bundle-version="[3.2.0,4.0.0)"
+ org.eclipse.jface;bundle-version="[3.2.0,4.0.0)",
+ org.eclipse.swtbot.swt.finder;bundle-version="4.2.1"
```
Comment (azoitl) [inline org.eclipse.draw2d.tests/META-INF/MANIFEST.MF:15]: I guess greater or equal to 4.2.1 would be better here or?

File: org.eclipse.draw2d.tests/META-INF/MANIFEST.MF (hunk 1 above)
Comment (ptziegler) [reply org.eclipse.draw2d.tests/META-INF/MANIFEST.MF]: > shouldn't this be 3.16.0 and should we really be so explicit?

I think I've raised the version number after adding it as a dependency. But you're right. For dependencies within the same project, the version constraint is probably not needed...

File: org.eclipse.draw2d.tests/META-INF/MANIFEST.MF (hunk 2 above)
Comment (ptziegler) [reply org.eclipse.draw2d.tests/META-INF/MANIFEST.MF:15]: Isn't that exactly what's described here?

image

Comment (azoitl) [reply org.eclipse.draw2d.tests/META-INF/MANIFEST.MF:15]: I always thought for ranges you need some range qualifier. Just checked on the OSGI spec and learned I was wrong. Really need to deepen my OSGI knowledge. An OSGI book is already on my reading list. Recommendations for more are warmly welcome.

**Review by azoitl — APPROVED** [review]

**Pull request merged**